ql/src/java/org/apache/hadoop/hive/ql/optimizer/StatsOptimizer.java

```diff
@@ -397,6 +397,11 @@ else if (getGbyKeyType(cgbyOp) == GbyKeyType.CONSTANT && rowCnt == 0) {
           if (udaf instanceof GenericUDAFSum) {
             // long/double/decimal
             ExprNodeDesc desc = aggr.getParameters().get(0);
+            // return null for SUM(1), when the table is empty. Without this, category = LONG, and the result is 0
+            // instead of NULL.
+            if (desc instanceof ExprNodeConstantDesc && rowCnt == 0) {
+              return null;
```

Comment:
if you could add a `null` to the result - then the whole computation would be removed
there is a chance that `oneRow.add(null)` will work

Reply:
Do you mean, instead of returning null, I should return oneRow.add(null) ?

Reply:
yes; I think that will work - but make sure to not execute further this conditional :)